manifests/charts/gateway/templates/deployment.yaml

```diff
@@ -65,8 +65,10 @@ spec:
             allowPrivilegeEscalation: false
```

Comment:
The options seem to be

1. Don't allow custom securityContext at all, and have  an `openshift` conditional.
2. Allow custom securityContext, and put our defaults in the default `values.yaml` (meaning people can break things with careless overrides)
3. Use a templating func  to force-overwrite whatever the custom securityContext is with the key-values we "must" have.


Of those 3 seems the best, for better or for worse we have a hardcoded depedency on the nonconfigurable value `1337` so we should probably always override any user-supplied `runAsUser`/`runAsGroup` keyvalues.